lib/workers/repository/update/branch/auto-replace.spec.ts

```diff
@@ -1038,5 +1038,284 @@ describe('workers/repository/update/branch/auto-replace', () => {
+        '{{depName}}@{{#if newDigest}}{{newDigest}}{{#if newValue}} # {{newValue}}{{/if}}{{/if}}{{#unless newDigest}}{{newValue}}{{/unless}}';
+      upgrade.depName = 'actions/checkout';
+      upgrade.currentValue = 'v1.0.0';
+      upgrade.currentDigest = undefined;
+      upgrade.depIndex = 0;
+      upgrade.pinDigests = true;
+      upgrade.updateType = 'replacement';
+      upgrade.replaceString = 'actions/checkout@v1.0.0';
+      upgrade.newValue = 'v2.0.0';
+      upgrade.newDigest =
+        '1cf887de9e8df1530de6a3a80e5fb8cef3c26c5248611921bd860c85313a688f';
+      upgrade.packageFile = 'workflows/build.yml';
+      const res = await doAutoReplace(
+        upgrade,
+        githubaction,
+        reuseExistingBranch
+      );
+      expect(res).toBe(
+        codeBlock`
+          name: build
+
+          on: [push]
+
+          jobs:
+            build:
+              runs-on: ubuntu-latest
+              steps:
+                - uses: actions/checkout@1cf887de9e8df1530de6a3a80e5fb8cef3c26c5248611921bd860c85313a688f # v2.0.0
+        `
+      );
+    });
+
+    it('github-actions: updates with pinDigest enabled but no currentDigest value', async () => {
+      const githubaction = codeBlock`
+        name: build
+
+        on: [push]
+
+        jobs:
+          build:
+            runs-on: ubuntu-latest
+            steps:
+              - uses: actions/checkout@v1.0.0
+      `;
+      upgrade.manager = 'github-actions';
+      upgrade.autoReplaceStringTemplate =
+        '{{depName}}@{{#if newDigest}}{{newDigest}}{{#if newValue}} # {{newValue}}{{/if}}{{/if}}{{#unless newDigest}}{{newValue}}{{/unless}}';
+      upgrade.depName = 'actions/checkout';
+      upgrade.currentValue = 'v1.0.0';
+      upgrade.currentDigest = undefined;
+      upgrade.depIndex = 0;
+      upgrade.pinDigests = true;
+      upgrade.updateType = 'replacement';
+      upgrade.replaceString = 'actions/checkout@v1.0.0';
+      upgrade.newName = 'some-other-action/checkout';
+      upgrade.newValue = 'v2.0.0';
+      upgrade.newDigest =
+        '1cf887de9e8df1530de6a3a80e5fb8cef3c26c5248611921bd860c85313a688f';
+      upgrade.packageFile = 'workflows/build.yml';
+      const res = await doAutoReplace(
+        upgrade,
+        githubaction,
+        reuseExistingBranch
+      );
+      expect(res).toBe(
+        codeBlock`
+          name: build
+
+          on: [push]
+
+          jobs:
+            build:
+              runs-on: ubuntu-latest
+              steps:
+                - uses: some-other-action/checkout@v2.0.0
+        `
+      );
+    });
+
+    it('github-actions: updates with pinDigest enabled and a currentDigest value', async () => {
+      const githubaction = codeBlock`
+        name: build
+
+        on: [push]
+
+        jobs:
+          build:
+            runs-on: ubuntu-latest
+            steps:
+              - uses: actions/checkout@2485f4d55aae6c5b073114bc4c4b1907c0abae14166281beee7d93f76ebf41fc # tag=v1.0.0
```

Comment:
why are the git sha so long? can we shorten them for better readability?

Reply:
Because the github-actions manager validates either [sha](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/github-actions/extract.ts#L17) or [shaShort](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/github-actions/extract.ts#L18)

Reply:
Done.  In doing so, I uncovered that the auto-replace didn't currently support digestShort updates.  This is now resolved too